core/src/main/kotlin/org/evomaster/core/search/service/ExtraHeuristicsLogger.kt

```diff
@@ -21,45 +28,61 @@ class ExtraHeuristicsLogger {
     @PostConstruct
     private fun postConstruct() {
         if (config.writeExtraHeuristicsFile) {
-
-            val path = getFilePath()
-
-            Files.createDirectories(path.parent)
-
-            Files.deleteIfExists(path)
-            Files.createFile(path)
-
-            path.toFile().appendText("testCounter,actionId,value,id,type,objective,group\n")
+            val extraHeuristicsFilePath = getExtraHeuristicsFilePath()
+            setUpFilePath(extraHeuristicsFilePath)
+            extraHeuristicsFilePath.toFile().appendText("testCounter,actionId,value,id,type,objective,group\n")
         }
     }
 
-    private fun getFilePath() = Paths.get(config.extraHeuristicsFile).toAbsolutePath()
+    /**
+     * Creates parent path if it is not exists.
+     * And deletes the file if the file already exists.
+     */
+    private fun setUpFilePath(filePath: Path) {
+        Files.createDirectories(filePath.parent)
+        Files.deleteIfExists(filePath)
+        Files.createFile(filePath)
+    }
 
+    private fun getExtraHeuristicsFilePath() = Paths.get(config.extraHeuristicsFile).toAbsolutePath()
 
     fun writeHeuristics(list: List<HeuristicEntryDto>, actionId: Int) {
-        if (!config.writeExtraHeuristicsFile) {
-            return
-        }
+        if (config.writeExtraHeuristicsFile) {
 
-        val counter = time.evaluatedIndividuals
+            val counter = time.evaluatedIndividuals
+            val lines = list.map {
+                val group = if (it.type == HeuristicEntryDto.Type.SQL) {
+                    try {
+                        "\"${removeAllConstants(it.id)}\""
```

Comment:
as this is going to be put in a CSV file, we need to make sure that we properly handle any `,` character

Reply:
One question, what about "org.evomaster.client.java.controller.api.dto.HeuristicEntryDto#id" 
How the "," character is handled here? 
It looks as the same problem concerns both it.id and  removeAllConstants(it.id), isn't it?

Reply:
hi. yes, technically it is the same issue. maybe we haven't encountered it before. anyway, if issue happens, we ll see problem as soon as we try to load the generated CSV file, due to column mismatches.
should check how `,` commas are escaped in CSV